### PR TITLE
feat: update transaction service and add documentation

### PR DESCRIPTION
**Pull request by syauqijan**
Hi, please help me review this PR to dev. Thanks 🙏

### CODE REVIEW

File: src/transactions/transactions.controller.ts [hunk 1]
```diff
@@ -13,23 +13,154 @@ import { TransactionsService } from './transactions.service';
...
+                type: 'string',
+                format: 'date-time',
+                example: '2024-11-17T13:00:00Z',
+              },
+            },
+          },
+        },
+      },
+    },
+  })
```
Comment (afnanramadhan) [inline src/transactions/transactions.controller.ts:91]: please add @ApiBearerAuth() for every endpoint that use authentication

File: src/transactions/transactions.service.ts [hunk 2]
```diff
@@ -11,33 +11,52 @@ import { UsersService } from '../users/users.service';
...
-      createdAt: new Date().toISOString(),
-      updatedAt: new Date().toISOString(),
-    };
+    const transactions = data.map((transaction) => {
+      const transactionId = uuidv4();
+      return {
+        ...transaction,
+        transaction_id: transactionId,
+        user_id,
+        createdAt: new Date().toISOString(),
```
Comment (afnanramadhan) [inline src/transactions/transactions.service.ts]: be consistent, use snake case, don't forget to update api response too

File: src/transactions/dto/create-transaction.dto.ts [hunk 3]
```diff
@@ -1,43 +1,37 @@
...
 } from 'class-validator';
 import { ApiProperty } from '@nestjs/swagger';
 
 export class CreateTransactionDto {
-  @IsString()
-  @IsNotEmpty({ message: 'user_id is required' })
-  user_id: string;
-
   @IsString()
   @IsNotEmpty({ message: 'name is required' })
```
Comment (afnanramadhan) [inline src/transactions/dto/create-transaction.dto.ts]: change to 'title is required'

File: src/transactions/transactions.controller.ts [hunk 4]
```diff
@@ -13,23 +13,154 @@ import { TransactionsService } from './transactions.service';
...
+  create(@Body() payload: { user_id: string; data: CreateTransactionDto[] }) {
```
Comment (afnanramadhan) [inline src/transactions/transactions.controller.ts]: i think its better to change CreateTransactioDto instead using payload, you can check this reference
1. https://chatgpt.com/share/6741c844-83e8-8006-adf7-0853dd713b59
2. https://www.reddit.com/r/Nestjs_framework/comments/r1p9qc/how_to_expose_an_array_of_objects_using_a_dto/

**Review by afnanramadhan — CHANGES_REQUESTED** [review]

File: src/transactions/transactions.service.ts (hunk 2 above)
Comment (syauqijan) [reply src/transactions/transactions.service.ts]: owh sorry i wasn't careful, ok i've changed it

File: src/transactions/dto/create-transaction.dto.ts (hunk 3 above)
Comment (syauqijan) [reply src/transactions/dto/create-transaction.dto.ts]: 🫡🫡🫡

File: src/transactions/transactions.controller.ts (hunk 4 above)
Comment (syauqijan) [reply src/transactions/transactions.controller.ts]: woww nice idea, ok i've changed it, sungkem🙏🏻 senior dev

**Review by afnanramadhan — APPROVED** [review]
approved
image

**Pull request merged**